nibabel/loadsave.py

```diff
@@ -56,39 +58,15 @@ def guessed_image_type(filename):
     image_class : class
         Class corresponding to guessed image type
     """
-    froot, ext, trailing = splitext_addext(filename, ('.gz', '.bz2'))
-    lext = ext.lower()
-    try:
-        img_type = ext_map[lext]
-    except KeyError:
-        raise ImageFileError('Cannot work out file type of "%s"' %
-                             filename)
-    if lext in ('.mgh', '.mgz', '.par'):
-        klass = class_map[img_type]['class']
-    elif lext == '.mnc':
-        # Look for HDF5 signature for MINC2
-        # https://www.hdfgroup.org/HDF5/doc/H5.format.html
-        with ImageOpener(filename) as fobj:
-            signature = fobj.read(4)
-            klass = Minc2Image if signature == b'\211HDF' else Minc1Image
-    elif lext == '.nii':
-        with ImageOpener(filename) as fobj:
-            binaryblock = fobj.read(348)
-        ft = which_analyze_type(binaryblock)
-        klass = Nifti2Image if ft == 'nifti2' else Nifti1Image
-    else:  # might be nifti 1 or 2 pair or analyze of some sort
-        files_types = (('image', '.img'), ('header', '.hdr'))
-        filenames = types_filenames(filename, files_types)
-        with ImageOpener(filenames['header']) as fobj:
-            binaryblock = fobj.read(348)
-        ft = which_analyze_type(binaryblock)
-        if ft == 'nifti2':
-            klass = Nifti2Pair
-        elif ft == 'nifti1':
-            klass = Nifti1Pair
-        else:
-            klass = Spm2AnalyzeImage
-    return klass
+    warnings.warn('guessed_image_type is deprecated', DeprecationWarning)
```

Comment:
Actually, does it make sense to have these at all? At least on my system, DeprecationWarnings are silently ignored, so they won't be raised. If they won't be raised, the `np.deprecate` is a better alert.

I'm not sure it's good manners to change the filters in a library like nibabel, but an alternative is to change the ignore rule for DeprecationWarnings.